flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileMonitoringFunction.java

```diff
@@ -58,12 +63,14 @@
  */
 @Internal
 public class ContinuousFileMonitoringFunction<OUT>
-	extends RichSourceFunction<TimestampedFileInputSplit> implements Checkpointed<Long> {
+	extends RichSourceFunction<TimestampedFileInputSplit> implements CheckpointedFunction {
 
 	private static final long serialVersionUID = 1L;
 
 	private static final Logger LOG = LoggerFactory.getLogger(ContinuousFileMonitoringFunction.class);
 
+	private static final String FILE_MONITORING_STATE_NAME = "file-monitoring-state";
```

Comment:
Instead of this you could also have
```
private static final ListStateDescriptor<Long> STATE_DESCRIPTOR =
    new ListStateDescriptor<>("file-monitoring-state", LongSerializer.INSTANCE);
```

and then use this descriptor later directly instead of initialising with this field.

That's just a personal style nitpick. Your version is also fine. 😃